### ISSUE TITLE
Accessing "document" from user-code uses the wrong iframe context

### ISSUE THREAD

**Issue by silbinarywolf** [cypress-io/cypress#2995]
This is a issue created from: https://github.com/cypress-io/cypress/issues/318#issuecomment-446467331 as requested by @jennifer-shehane 

If I was going to continue with the project linked below, I'd work towards giving you guys more details, but I'm currently focusing on a Golang project in my downtime, so sorry that the information isn't as precise as it could be!

### Current behavior:
Apologies if this is wrong, but I'm going off my vague memory from a few weeks ago. The "test running" code is running in a <iframe> seperate to where the element is rendered, any non-test code that uses document.querySelector will not work as expected. To work around this, I simply changed my non-test code... but ideally I shouldn't have to! I want to be able to provide an API similar to https://flatpickr.js.org/, so being able to use document.querySelector would be good.

I can imagine this would cause issues for people who are trying to install Cypress on a legacy project that utilizes document.querySelector for whatever reason. (Mix of jQuery / VDOM-framework)

### Desired behavior:
Cypress should hopefully be able to make "document" in user code (ie. not test code) refer to the correct iframe.

### Steps to reproduce: (app code and test code)
https://github.com/silbinarywolf/collapo/blob/a6de78b1bbf336fa683514e44b5582579ddc272e/src/Collapo.ts#L26

### Versions
3.1.1

**Comment by silbinarywolf** [comment]
@jennifer-shehane Yep that's it :)

@brian-mann Yeah I also got the sense that proxying the document/window stuff would probably not be worth the complexity / cognitive overhead. It's just not clear how I can render/mount/test the component in a type-safe way with different states and options if I mount it via the app. ie. How do I visually see it being interacted with, button clicks, etc?

I feel like I should be able to import from the app iframe context within the spec, this would mean `document` would be the app document. However, I have no idea if this is possible and even if it is, I'd assume that doing so in Internet Explorer/etc would be impossible or be a huge pain.

Also as an FYI, I took inspiration from this:
- https://github.com/bahmutov/cypress-react-unit-test/blob/master/lib/index.js
